examples/Calibration/Calibration.ino

```diff
@@ -0,0 +1,668 @@
+/** 
+ * Calibration.ino
+ * Interactive calibration of input and output voltages of the Tsunami.
+ * Requires a generic digital multimeter for the output calibration phase
+ * and an output-to-input loopback connection for input calibration phase.
+ *
+ * Accepts commands via the serial port at 115200 baud.
+ */
+
+#include <SPI.h>
+#include <EEPROM.h>
+#include <tsunami.h>
+
+#define BUFFER_SIZE     32
+#define SAMPLE_SIZE     16
+#define NUM_MEASURE     32
+
+void setup() {
+  // Initialize the serial port
+  Serial.begin(115200);
+  
+  // Initialize the Tsunami
+  Tsunami.begin();
+}
+
+// Waits until a newline arrives from the serial port
+void waitLine() {
+  char key;
+  while(1) {
+    if(Serial.available()) {
+      key = Serial.read();
+
+      if(key == '\n')
+        return;
+    }
+  }
+}
+
+// Reads a line from the serial port, without timeout
+int readLine(char *buf, int len) {
+  int pos = 0;
+  while(1) {
+    if(Serial.available()) {
+      buf[pos++] = Serial.read();
+  
+      if(pos == len || buf[pos - 1] == '\n')
```

Comment:
I'm having a slight issue with reformatting this one: by necessity, "pos" always points to the next free buffer char, hence the test of buf[pos - 1]; here it works because the test is inside the "if" and only tested once pos > 0; however, outside in the "while" it would initially cause access to buf[-1], and I'm having a hard time avoiding that without piling on rather kludgey safeguards or exceptions to avoid it happening. I might be missing the obvious - do you see any way to do this elegantly...?